src/BuildIt.Forms/BuildIt.Forms.Controls/CameraPreviewControl.xaml.cs

```diff
@@ -106,6 +154,56 @@ public Aspect Aspect
         /// </summary>
         internal Func<Task<IReadOnlyList<ICamera>>> RetrieveCamerasFunc { get; set; }
 
+        /// <summary>
+        /// Start camera preview
+        /// </summary>
+        /// <returns></returns>
+        public async Task StartPreview()
+        {
+            if (StartPreviewFunc == null || Status == CameraStatus.Started || Status == CameraStatus.Starting)
+            {
+                return;
+            }
+
+            if (await startCameraPreviewSemaphoreSlim.WaitAsync(0))
+            {
+                try
+                {
+                    Status = CameraStatus.Starting;
+
+                    await StartPreviewFunc.Invoke();
+                }
+                finally
+                {
+                    startCameraPreviewSemaphoreSlim.Release();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Stop camera preview
+        /// </summary>
+        /// <returns></returns>
+        public async Task StopPreview()
+        {
+            if (StopPreviewFunc == null || Status == CameraStatus.Stopped || Status == CameraStatus.None)
+            {
+                return;
+            }
+
+            if (await stopCameraPreviewSemaphoreSlim.WaitAsync(0))
+            {
+                try
+                {
+                    await StopPreviewFunc.Invoke(CameraPreviewStopParameters.Default);
```

Comment:
We should capture the exception and handle that gracefully - we should have an error state for the camera

Reply:
Will address in future commits